### ISSUE TITLE
[BUG] mtp_<k> loss logging drifts from CP=1 baseline under THD+CP due to AVG-of-per-rank-ratio reduction

### ISSUE THREAD

**Issue by wplf** [NVIDIA/Megatron-LM#4896]
**Describe the bug**

`MTPLossLoggingHelper` reports per-layer MTP loss as the *unweighted average across DP×CP ranks of pre-divided per-rank ratios* rather than the token-weighted global mean. Whenever per-rank `num_tokens` are unequal — which is the norm under THD+CP (per-sample zigzag partition + `roll_tensor` boundary effects, magnified when `loss_mask` is sparse) — the reported value diverges from `Σ(loss) / Σ(tokens)`.

Gradients are unaffected (`process_mtp_loss` does correct token-count rescaling for the backward path); **only the logged value drifts**, which makes CP correctness experiments confusing because the LM loss aligns across CP configs while `mtp_<k> loss` does not.

Tag the [@mcore-oncall](https://github.com/orgs/NVIDIA/teams/mcore-oncall).

**Affected code**

1. `megatron/core/transformer/multi_token_prediction.py:723-731` — per-rank pre-divide:

```python
mtp_loss_for_log = (
    torch.sum(mtp_loss) * (num_tokens > 0).to(mtp_loss.dtype)
) / num_tokens.clamp(min=1)
MTPLossLoggingHelper.save_loss_to_tracker(
    mtp_loss_for_log,
    mtp_layer_number,
    config.mtp_num_layers,
    avg_group=parallel_state.get_data_parallel_group(with_context_parallel=True),
)
```

2. `megatron/core/transformer/multi_token_prediction.py:395-398` — AVG-of-ratio reduction:

```python
if tracker.get('avg_group') is not None:
    torch.distributed.all_reduce(
        values, group=tracker['avg_group'], op=torch.distributed.ReduceOp.AVG
    )
```

`AVG_ranks(Σ_r / N_r) ≠ Σ_ranks(Σ_r) / Σ_ranks(N_r)` whenever `N_r` differs across ranks. The direction of bias depends on the correlation between per-rank `num_tokens` and per-rank per-token loss, so the drift can be in either direction (~1–5% in our SFT-style runs).

**Steps/Code to reproduce bug**

Any THD+CP training with `--mtp-num-layers > 0`, comparing CP=1 vs CP>1 with otherwise identical config. The drift is amplified when `loss_mask` is sparse and concentrated (e.g. SFT mask that supervises only the assistant span).

Minimal example with one of the in-repo MoE configs:

```
TP=1 PP=1 EP=1 CP=1 GBS=16 --mtp-num-layers 1 --use-packed-sequence ...   # baseline
TP=1 PP=1 EP=1 CP=2 GBS=16 --mtp-num-layers 1 --use-packed-sequence ...   # test
```

Observed:
- `lm loss` aligns between CP=1 and CP=2 (uses sum-then-divide via `--calculate-per-token-loss`).
- `mtp_1 loss` does **not** align — typically off by ~1–5%.

**Expected behavior**

Reported `mtp_<k> loss` should equal the token-weighted global mean across all DP×CP ranks, so that two runs with same data + different CP report the same value (matching how `lm loss` already behaves with `--calculate-per-token-loss`).

**Suggested fix**

Track `(loss_sum, num_tokens)` per layer separately and divide post-reduction with `op=SUM`:

```python
# per-rank, in process_mtp_loss
loss_sum = torch.sum(mtp_loss)
MTPLossLoggingHelper.save_loss_to_tracker(
    loss_sum=loss_sum,
    num_tokens=num_tokens,
    layer_number=mtp_layer_number,
    num_layers=config.mtp_num_layers,
    reduce_group=parallel_state.get_data_parallel_group(with_context_parallel=True),
)

# in reduce_loss_in_tracker: SUM both tensors, then divide
torch.distributed.all_reduce(loss_sums,    op=torch.distributed.ReduceOp.SUM, group=...)
torch.distributed.all_reduce(token_counts, op=torch.distributed.ReduceOp.SUM, group=...)
reported = loss_sums / token_counts.clamp(min=1)
```

This is the same reduction pattern `forward_step.loss_func` already uses for the main LM loss path with `--calculate-per-token-loss`.

**Additional context**

- Related but distinct from #3943 and #1532 — both addressed gradient scaling (`MTPLossAutoScaler`), not the logging path. The MTP gradient correction in `process_mtp_loss` (lines 732+) is already correct under CP; this issue is purely about the value reported to TB/W&B.
- We hit this while CP-correctness-testing an SFT setup (sparse `loss_mask` concentrated at the end of each sample). With a denser mask the per-rank `num_tokens` are roughly balanced and the drift is small; with the sparse SFT mask the bias becomes visible (~3% in our 0.8B THD+CP=2 run).
